# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/procps/hotfix-glibc236.patch
# Copyright (C) 2022 - 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- procps-ng-4.0.3/src/pgrep.c.orig	2023-05-14 23:38:49.482807733 +0200
+++ procps-ng-4.0.3/src/pgrep.c	2023-05-14 23:40:21.961814411 +0200
@@ -42,13 +42,17 @@
 #include <stdbool.h>
 #include <time.h>
 
-#ifdef ENABLE_PIDWAIT
+#ifdef ENABLE_PIDWAIT && !defined(HAVE_PIDFD_OPEN)
 #include <sys/epoll.h>
 #ifndef HAVE_PIDFD_OPEN
 #include <sys/syscall.h>
 #endif /* !HAVE_PIDFD_OPEN */
 #endif
 
+#if defined(HAVE_PIDFD_OPEN)
+#include <sys/pidfd.h>
+#endif
+
 /* EXIT_SUCCESS is 0 */
 /* EXIT_FAILURE is 1 */
 #define EXIT_USAGE 2
